reproject/interpolation/core_full.py

```diff
@@ -126,10 +127,22 @@ def _reproject_full(array, wcs_in, wcs_out, shape_out, order=1):
 
     coordinates = pixel_in.transpose()[::-1]
 
-    array_new = map_coordinates(array,
+    if array_out is not None:
+        assert array_out.shape == shape_out
+        array_new = array_out
+        array_new[:] = map_coordinates(array,
                                 coordinates,
                                 order=order, cval=np.nan,
                                 mode='constant'
                                 ).reshape(shape_out)
-
-    return array_new, (~np.isnan(array_new)).astype(float)
+    else:
+        array_new = map_coordinates(array,
+                                    coordinates,
+                                    order=order, cval=np.nan,
+                                    mode='constant'
+                                    ).reshape(shape_out)
```

Comment:
I found a better solution.  `map_coordinates` should dump into an array which will then be copied into the `array_out` array if one is specified.